t/28_num.t

```diff
@@ -0,0 +1,38 @@
+use strict;
+use Test::More 0.98;
+use Test::Exception;
+use Capture::Tiny ':all';
+use Getopt::Kingpin;
+
+
+subtest 'num normal' => sub {
+    local @ARGV;
+    push @ARGV, qw(--x 0);
+
+    my $kingpin = Getopt::Kingpin->new;
+    my $x = $kingpin->flag('x', 'set x')->num();
+    $kingpin->parse;
+
+    is $x->value, 0;
+    ok $x == 0;
+};
+
```

Comment:
Please add a test for `looks_like_number` cases.
For example, `1e3` or `-1.5e6`.